### DISCUSSION TITLE
Need a router based on configuration so i can dynamically add and remove

### DISCUSSION THREAD

**Discussion by sathish2k** [vercel/next.js#71553] [App Router]
# Router Configuration

You can define your routes using a configuration object, similar to Angular or React Router. Below is an example of how to set it up:

```javascript
const routes = [
  {
    id: "root",
    path: "/",
    layout: RootLayout,
    children: [
      {
        id: "dashboard",
        path: "/dashboard",
        component: Dashboard,
      },
      {
        id: "recent-blog",
        path: "/recent-blog",
        component: RecentBlog,
      },
    ],
  },
];

**Comment by icyJoseph** [comment]
> Next.js uses a file-system based router where folders are used to define routes.

**Reply by sathish2k** [reply]
Yeah I know but i'm trying something routes based on config, it can be achieved through dynamic route but still i have doubt like will layout, error, loading etc can be achieved or not

**Reply by icyJoseph** [reply]
I mean, in pages router it could easily be done with dynamic route, some people even used react-router-dom on top of Next.js, which is... well yeah, not good.

I think, while it might be a good exercise in learning, this is not something to take to production. 

> Layout do not re-render on navigation, therefore are not recommend for auth checks. Consider a shared navigation that displays user information:

- https://github.com/vercel/next.js/discussions/63775
